### ISSUE TITLE
Unable to debug Blazor hosted webassembly 3.2.0 from Visual Studio 2019 16.6.2

### ISSUE THREAD

**Issue by spapaseit** [dotnet/aspnetcore#22911]
### Describe the bug
I have Visual Studio 2019 16.6.2 installed, along with the .NET Core 3.1.301 SDK and all the necessary Blazor templates.

If I create a brand new Blazor WebAssmbly application, I can run it from Visual Studio by pressing <kbd>Ctrl</kbd>+<kbd>F5</kbd> and it opens a new tab on my active browser window and starts fine.  

But when I press <kbd>F5</kbd>, a new, separate browser window is opened instead of a tab in my main browser window (which is a pity), but the page remains at `about:blank` for a while until an error message appears in Visual Studio:

```
Failed to launch debug adapter.  Additional information may be available in the output window.

Unable to launch browser: 
"Could not open wss://localhost:5001/_framework/debug/ws-proxy?browser=wsAFFlocalhostA50821FdevtoolsFbrowserFceed3b95-58ac-470a-b10c-5d139cfd7117": timeout after 60000ms
```

Whether the Blazor WebAssembly application is "Hosted" or not doesn't make a difference.

I've followed the steps in the [Enforce HTTPS in ASP.NET Core](https://docs.microsoft.com/en-gb/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.1&tabs=visual-studio#troubleshoot-certificate-problems), but that made no difference either.

If, while waiting for the `about:blank` window to crash, I manually open the client side url on a browser tab, the app loads and the debugger breakpoints are hit.
During the short while before it goes on to crash, if I hit <kbd>Shift</kbd>+<kbd>Alt</kbd>+<kbd>D</kbd> I *always* get the following error:

```
Unable to find debuggable browser tab
Could not get a list of browser tabs from http://127.0.0.1:9222/json. Ensure your browser is running with debugging enabled.

Resolution

If you are using Microsoft Edge (80+) for your development, follow these instructions:
Press Win+R and enter the following:

msedge --remote-debugging-port=9222 --user-data-dir="C:\Users\[my_username]\AppData\Local\Temp\blazor-edge-debug" --no-first-run https://localhost:44372/
```

I get this each and every time, no matter how often I run the command (having closed all browser instances, rebooted, etc)

This happens in both Edge 84 (Edgium) and Chrome.

I should be able to create a new Blazor WebAssembly app and hit <kbd>F5</kbd> on Visual Studio out of the box. What am I missing?

#### =============== UPDATE ==================
I've just tried using "old Edge" (EdgeHtml) and Firefox as the debug browser in Visual Studio, and both work as expected (meaning that the application at least starts up when pressing <kbd>F5</kbd>). The problem apears to be both Chromium-based browsers then...

#### =============== UPDATE 2 ==================
Removing the `IIS Express` section from `launchSettings.json` on the Server project allows VS 2019 to start Kestrel up and that does load the Blazor Wasm app on a new browser window. <kbd>Shift</kbd>+<kbd>Alt</kbd>+<kbd>D</kbd> still fails with `Unable to find debuggable browser tab`, though.

### To Reproduce

Just create a fresh Blazor WebAssembly from Visual Studio and press <kbd>F5</kbd>

### Further technical details
- ASP.NET Core 3.1.3
- Visual Studio 2019 16.6.2
- Windows 10 (1909)

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us.
We're moving this issue to the `Next sprint planning` milestone for future evaluation / consideration. We will evaluate the request when we will planning the work for the next milestone. To learn more about what to expect next and how this issue will be handled you can read more about our triage process [here](https://github.com/dotnet/aspnetcore/blob/master/docs/TriageProcess.md).

**Comment by davidmilligan** [comment]
I'm having a very similar issue, with exactly the same symptoms, but in my case, a new blank wasm project actually works. It's just my own existing project that I can't get working. I've tried making everything as similar as possible between the two, but it just doesn't work.

![image](https://user-images.githubusercontent.com/5251903/85412769-f49d4e00-b537-11ea-987f-bedde1cfa27d.png)

I also get the same error message when trying the  <kbd>Shift</kbd>+<kbd>Alt</kbd>+<kbd>D</kbd>

**Comment by kollfeldt** [comment]
@spapaseit had the exact same issue, resolved it by uninstalling Hyper-V. Hopefully it can help you.

**Comment by spapaseit** [comment]
@kollfeldt, thanks for the tip, but unfortunately uninstalling Hyper-V (and restarting) made no difference for me; still having the exact same issue.

**Comment by bignorm64** [comment]
I too have a similar issue. F5 opens the browser and it sits spinning on about:blank and if left will timeout as above. Pressing F12 before the timeout to enter the browser developer tools triggers the blazor webassembly app to load but almost without exception no breakpoints are hit.

**Comment by RealHustengutzel** [comment]
had also the same issue. I enabled ScriptDebugging(dropdown at Start debugging) in VisualStudio and now it works fine.
![image](https://user-images.githubusercontent.com/68413981/87760397-fe049800-c80f-11ea-9e65-919fd2cc0f6a.png)

**Comment by spapaseit** [comment]
Thanks for the tip, @RealHustengutzel. Unfortunately, it made no difference in my case either.

**Comment by NPadrutt** [comment]
I have the same issue, funny enough only on my work pc. When I try it on another one it works as expected.

**Comment by lnodaba** [comment]
I have the same issue, finally instead of IIS Express I use the project launchsettings:

![image](https://user-images.githubusercontent.com/38653825/88255977-9395a100-ccc2-11ea-824e-726056c91cd5.png)

**Comment by hamdand** [comment]
Hi,
After searching for couple of days, I did the following and it worked like charm:
1.Install IE "Microsoft edge"
2.Use IE "Microsoft edge" as your VS. browser (*NOT CHROME*)
3.Use your application as server (*NOT IIS EXPRESS*).
Enjoy.
![image](https://user-images.githubusercontent.com/6485997/88262727-cc8d4000-ccd9-11ea-9158-a8248cdcce90.png)

**Comment by captainsafia** [comment]
Hello folks,

There's quite a few different issues reported here. Some of them have been previously reported and will be resolved in the upcoming .NET 5/VS 16.8 release. Some things to note:

- There are some issues with reliably launching a Blazor application when using IIS. We expect these will be solved in VS 16.8.
- The timeouts occur when the the debugger takes too long to start. We've made some improvements to this that will ship in the same timeline.
- Some issues with unbound exceptions make occur. If that is the case, make sure that the you've accepted local development certificates if you are using an HTTPS-based domain.

I'd recommend trying the latest previews of .NET 5 and VS 16.8 and sharing feedback there.

**Comment by proskilly** [comment]
SO I can't debug blazor webassembly web apps and you are saying it isn't going to be fixed till November?

**Comment by proskilly** [comment]
I hope this helps someone.
Whenever I ran my blazor webassembly hosted app in Visual Studio the browser opened but it just stayed on about:blank for ages and never loaded the web page/app.
I found that if you open up the browser dev tools (Ctrl + Shift + I) it shoved it along and the page actually loaded and it all worked. This works in both Chrome and Edge (at least for me).

**Comment by lnodaba** [comment]
@proskilly you mean in the meantime you're waiting on the browser which hangs you just simply open up the console and it suddenly loads?

**Comment by proskilly** [comment]
@lnodaba  Yes that's right.

**Comment by Will4000** [comment]
Hi All,
Long time listener, first time caller.
A bit disappointing about resolution timeframe, here's what worked for me...
I found this file:
debuggerConfiguration-1859ae3b-3952-46c9-8065-7b07286175be.json
under ....\MyApp\.vs\MyApp\config
it contains a single line
  {"chromiumDebuggerPort":60128}
The file was a couple of weeks old, which struck me as odd, anyhow i deleted the file.
Then on next run-with-debug a new you-beaut file was created and my debugging came good.
My guess is that VS is having trouble exchanging that file, especially when changing Debug properties for the project.
FWIW - this is a hosted Blazor web-assembly application

Aside from some esoteric bits and bobs, i am loving Blazor and the VS2019 environment.

Go hard MS.

W.

**Comment by captainsafia** [comment]
Thanks for sharing those details, @Will4000. As mentioned, we've been working on a variety of debugging improvements that will ship with .NET5/16.8. To help with triage/resolution, it would be great to open a new issue for each problem you want to report.

This will make it easier for us to get any necessary logs/details for your scenario and get to a root cause faster.

**Comment by TorrentialFire** [comment]
> I hope this helps someone.
> Whenever I ran my blazor webassembly hosted app in Visual Studio the browser opened but it just stayed on about:blank for ages and never loaded the web page/app.
> I found that if you open up the browser dev tools (Ctrl + Shift + I) it shoved it along and the page actually loaded and it all worked. This works in both Chrome and Edge (at least for me).

This. So much. Thank you for that tip! I was running a fresh install of .NET 5 Preview 8, the latest VS Code, and a fresh install of google Chrome and getting when trying to launch using a "blazorwasm" type launch config in VS Code. `Ctrl + Shift + I` kicks something into gear in Chrome and the error is no longer thrown. I can debug from VS Code and Visual Studio now!

**Comment by spapaseit** [comment]
Hi @captainsafia, I finally got round to uinstalling VS 2019 16.8 and Preview 8 of .NET 5, but the issue still persists; the behaviour is still exactly the same, even with a fresh Hosted Blazor WebAssembly project.

**Comment by rakeshkgsg** [comment]
> Hi @captainsafia, I finally got round to installing VS 2019 16.8 and Preview 8 of .NET 5, but the issue still persists; the behaviour is still exactly the same, even with a fresh Hosted Blazor WebAssembly project.

Yes i have the similar issue with the latest .net5 previrw 8. Strangely., when i create a new project, it works and i can debug the code. Then i stop and start it again. It will have the forever spin. I open a new chrome or ie edge and it works or if i press f12 to debug the page shows but cant be debugged. Sometimes, after waiting for a long tome and then when i debug again, it starts working. I deleted the bin and obj folder also to no avail.

**Comment by captainsafia** [comment]
@rakeshkgsg @spapaseit Can you share your exact version of VS that you are using? You should be able to access it by clicking on the "Preview" label in the top-right of the Visual Studio window.

Also, if you code share the contents of the `%TEMP%vscode-chrome-debug.txt` file after a failed debugging session that would be great?

**Comment by Murilobdo** [comment]
I have the same problem on the loading page about: blank and without a definite answer, for more details on my problem at 
#25566

**Comment by spapaseit** [comment]
@captainsafia, sorry for the delay on this. I'm using `Microsoft Visual Studio Professional 2019 Preview
Version 16.8.0 Preview 2.1`

 
![image](https://user-images.githubusercontent.com/4639889/92323577-2d7a9880-f03a-11ea-8da3-9501e5c1a6ab.png)


### `vscode-edge-debug.txt`

```
{"timestamp":1598898568353,"tag":"runtime.welcome","level":1,"message":"js-debug v1.48.1 started","metadata":{"os":"win32 x64","nodeVersion":"v12.16.3","adapterVersion":"1.48.1"}}
{"tag":"dap.receive","timestamp":1598898568193,"metadata":{"connectionId":0,"message":{"type":"request","command":"initialize","arguments":{"pathFormat":"path","clientID":"visualstudio","clientName":"Visual Studio","adapterID":"pwa-msedge","locale":"en-US","linesStartAt1":true,"columnsStartAt1":true,"supportsVariableType":true,"supportsRunInTerminalRequest":true,"supportsMemoryReferences":true,"supportsProgressReporting":true,"SupportsMessageBox":true,"supportsHandshakeRequest":true,"supportsVsAdditionalBreakpointBinds":true,"supportsVsCustomMessages":true,"supportsMapURLToFilePathRequest":true,"supportsLaunchUnelevatedProcessRequest":true},"seq":1}},"level":0}
{"tag":"dap.send","timestamp":1598898568195,"metadata":{"connectionId":0,"message":{"seq":1,"type":"response","request_seq":1,"command":"initialize","success":true,"body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":false,"supportsConditionalBreakpoints":true,"supportsHitConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"exceptionBreakpointFilters":[{"filter":"caught","label":"Caught Exceptions","default":false},{"filter":"uncaught","label":"Uncaught Exceptions","default":false}],"supportsStepBack":false,"supportsSetVariable":true,"supportsRestartFrame":true,"supportsGotoTargetsRequest":false,"supportsStepInTargetsRequest":false,"supportsCompletionsRequest":true,"supportsModulesRequest":false,"additionalModuleColumns":[],"supportedChecksumAlgorithms":[],"supportsRestartRequest":true,"supportsExceptionOptions":false,"supportsValueFormattingOptions":true,"supportsExceptionInfoRequest":true,"supportTerminateDebuggee":false,"supportsDelayedStackTraceLoading":true,"supportsLoadedSourcesRequest":true,"supportsLogPoints":true,"supportsTerminateThreadsRequest":false,"supportsSetExpression":false,"supportsTerminateRequest":false,"completionTriggerCharacters":[".","[","\"","'"],"supportsBreakpointLocationsRequest":true,"supportsClipboardContext":true}}},"level":0}
{"tag":"dap.send","timestamp":1598898568197,"metadata":{"connectionId":0,"message":{"seq":2,"type":"event","event":"initialized","body":{}}},"level":0}
{"tag":"dap.receive","timestamp":1598898568339,"metadata":{"connectionId":0,"message":{"type":"request","command":"launch","arguments":{"WebRoot":"C:\\Software Development\\Projects\\CodeNamed\\decontract.api\\src\\Decontract.Server\\wwwroot","rootPath":"C:\\Software Development\\Projects\\CodeNamed\\decontract.api\\src\\Decontract.Server\\wwwroot","__workspaceFolder":"C:\\Software Development\\Projects\\CodeNamed\\decontract.api\\src\\Decontract.Server\\wwwroot","runtimeExecutable":"C:\\Program Files (x86)\\Microsoft\\Edge Beta\\Application\\msedge.exe","runtimeArgs":[],"sourceMaps":true,"url":"https://localhost:44363/","trace":{"logFile":"C:\\Users\\sergi\\AppData\\Local\\Temp\\vscode-edge-debug.txt"},"port":60071,"launchUnelevated":true,"projectGuid":"29768966-02c6-4faf-80e6-7d0e337f3bd9","userDataDir":"C:\\Users\\sergi\\AppData\\Local\\Microsoft\\VisualStudio\\16.0_9c289d2a\\WebTools\\B18EC5CF_E846888C","breakOnLoad":true,"pathMapping":{"/_content/ChartJs.Blazor":"C:\\Users\\sergi\\.nuget\\packages\\chartjs.blazor\\2.0.0-preview1\\build\\..\\staticwebassets\\","/_content/Microsoft.AspNetCore.Components.WebAssembly.Authentication":"C:\\Users\\sergi\\.nuget\\packages\\microsoft.aspnetcore.components.webassembly.authentication\\5.0.0-preview.8.20414.8\\build\\..\\staticwebassets\\"},"smartStep":true,"skipFiles":[],"showAsyncStacks":true,"timeout":60000,"inspectUri":"{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}","cleanUp":"onlyTab","name":"Visual Studio launch configuration override","type":"pwa-msedge","request":"launch","continueOnDisconnect":true},"seq":2}},"level":0}
{"tag":"dap.send","timestamp":1598898568341,"metadata":{"connectionId":0,"message":{"seq":3,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/launch","data":{"type":"pwa-msedge","request":"launch","os":"win32 x64","nodeVersion":"v12.16.3","adapterVersion":"1.48.1","parameters":"{\"type\":\"pwa-msedge\",\"name\":\"<string>\",\"request\":\"launch\",\"trace\":{\"logFile\":\"<string>\"},\"outputCapture\":\"console\",\"timeout\":60000,\"timeouts\":{},\"showAsyncStacks\":true,\"skipFiles\":[],\"smartStep\":true,\"sourceMaps\":true,\"pauseForSourceMap\":true,\"resolveSourceMapLocations\":null,\"rootPath\":\"<string>\",\"outFiles\":[\"<string>\",\"<string>\"],\"sourceMapPathOverrides\":{\"webpack://?:*/*\":\"<string>\",\"webpack:///./~/*\":\"<string>\",\"meteor://💻app/*\":\"<string>\"},\"enableContentValidation\":true,\"__workspaceFolder\":\"<string>\",\"__autoExpandGetters\":false,\"address\":\"localhost\",\"port\":60071,\"disableNetworkCache\":true,\"pathMapping\":{\"/_content/ChartJs.Blazor\":\"<string>\",\"/_content/Microsoft.AspNetCore.Components.WebAssembly.Authentication\":\"<string>\"},\"url\":\"<string>\",\"restart\":false,\"urlFilter\":\"\",\"webRoot\":\"<string>\",\"server\":null,\"targetSelection\":\"automatic\",\"vueComponentPaths\":[\"<string>\",\"<string>\"],\"cwd\":null,\"file\":null,\"env\":{},\"includeDefaultArgs\":true,\"runtimeArgs\":[],\"runtimeExecutable\":\"<string>\",\"userDataDir\":\"<string>\",\"browserLaunchLocation\":\"workspace\",\"profileStartup\":false,\"cleanUp\":\"<string>\",\"useWebView\":false,\"WebRoot\":\"<string>\",\"launchUnelevated\":true,\"projectGuid\":\"<string>\",\"breakOnLoad\":true,\"inspectUri\":\"<string>\",\"continueOnDisconnect\":true}"}}}},"level":0}
{"tag":"dap.send","timestamp":1598898568344,"metadata":{"connectionId":0,"message":{"seq":4,"type":"event","event":"output","body":{"category":"console","output":"Verbose logs are written to:\nC:\\Users\\sergi\\AppData\\Local\\Temp\\vscode-edge-debug.txt\n"}}},"level":0}
{"tag":"dap.send","timestamp":1598898568356,"metadata":{"connectionId":0,"message":{"seq":5,"type":"request","command":"launchUnelevated","arguments":{"process":"C:\\Program Files (x86)\\Microsoft\\Edge Beta\\Application\\msedge.exe","args":["--disable-background-networking","--disable-background-timer-throttling","--disable-backgrounding-occluded-windows","--disable-breakpad","--disable-client-side-phishing-detection","--disable-default-apps","--disable-dev-shm-usage","--disable-renderer-backgrounding","--disable-sync","--metrics-recording-only","--no-first-run","--no-default-browser-check","--user-data-dir=C:\\Users\\sergi\\AppData\\Local\\Microsoft\\VisualStudio\\16.0_9c289d2a\\WebTools\\B18EC5CF_E846888C","--remote-debugging-port=60071","about:blank"]}}},"level":0}
{"tag":"dap.receive","timestamp":1598898568528,"metadata":{"connectionId":0,"message":{"type":"request","command":"setBreakpoints","arguments":{"source":{"path":"c:\\Software Development\\Projects\\CodeNamed\\decontract.api\\src\\Decontract.Web\\Pages\\DashboardComponentBase.cs","sources":[],"checksums":[]},"breakpoints":[{"line":26,"column":17}],"lines":[26]},"seq":3}},"level":0}
{"tag":"dap.send","timestamp":1598898568528,"metadata":{"connectionId":0,"message":{"seq":6,"type":"response","request_seq":3,"command":"setBreakpoints","success":true,"body":{"breakpoints":[{"id":1,"verified":false,"message":"Unbound breakpoint"}]}}},"level":0}
{"tag":"dap.receive","timestamp":1598898568560,"metadata":{"connectionId":0,"message":{"type":"request","command":"setBreakpoints","arguments":{"source":{"path":"c:\\Software Development\\Projects\\CodeNamed\\decontract.api\\src\\Decontract.Server\\Features\\Dashboard\\DashboardController.cs","sources":[],"checksums":[]},"breakpoints":[{"line":33,"column":17}],"lines":[33]},"seq":4}},"level":0}
{"tag":"dap.receive","timestamp":1598898568560,"metadata":{"connectionId":0,"message":{"type":"request","command":"setBreakpoints","arguments":{"source":{"path":"c:\\Software Development\\Projects\\CodeNamed\\decontract.api\\src\\Decontract.Web\\Components\\Widgets\\DocumentsByDepartment.razor","sources":[],"checksums":[]},"breakpoints":[{"line":83,"column":13}],"lines":[83]},"seq":5}},"level":0}
{"tag":"dap.send","timestamp":1598898568560,"metadata":{"connectionId":0,"message":{"seq":7,"type":"response","request_seq":4,"command":"setBreakpoints","success":true,"body":{"breakpoints":[{"id":2,"verified":false,"message":"Unbound breakpoint"}]}}},"level":0}
{"tag":"dap.send","timestamp":1598898568560,"metadata":{"connectionId":0,"message":{"seq":8,"type":"response","request_seq":5,"command":"setBreakpoints","success":true,"body":{"breakpoints":[{"id":3,"verified":false,"message":"Unbound breakpoint"}]}}},"level":0}
{"tag":"dap.receive","timestamp":1598898568663,"metadata":{"connectionId":0,"message":{"type":"response","request_seq":5,"success":true,"command":"launchUnelevated","body":{},"seq":6}},"level":0}
{"tag":"dap.receive","timestamp":1598898568752,"metadata":{"connectionId":0,"message":{"type":"request","command":"setExceptionBreakpoints","arguments":{"filters":[]},"seq":7}},"level":0}
{"tag":"dap.send","timestamp":1598898568752,"metadata":{"connectionId":0,"message":{"seq":9,"type":"response","request_seq":7,"command":"setExceptionBreakpoints","success":true,"body":{}}},"level":0}
{"tag":"dap.receive","timestamp":1598898568762,"metadata":{"connectionId":0,"message":{"type":"request","command":"configurationDone","arguments":{},"seq":8}},"level":0}
{"tag":"dap.send","timestamp":1598898568763,"metadata":{"connectionId":0,"message":{"seq":10,"type":"response","request_seq":8,"command":"configurationDone","success":true,"body":{}}},"level":0}
{"tag":"runtime.launch","timestamp":1598898569534,"message":"Discovered target URL from /json/version","metadata":{"url":"ws://localhost/devtools/browser/aa4ab88b-d911-48fd-b6d5-94411397049c","fixed":"ws://localhost:60071/devtools/browser/aa4ab88b-d911-48fd-b6d5-94411397049c"},"level":0}
{"tag":"dap.send","timestamp":1598898573202,"metadata":{"connectionId":0,"message":{"seq":11,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/dap/operation","data":{"errors":[],"initialize":{"operation":"initialize","totalTime":3.8,"max":3.8,"avg":3.8,"stddev":null,"count":1,"failed":0},"!initialize.errors":[],"setBreakpoints":{"operation":"setBreakpoints","totalTime":2.1,"max":0.7,"avg":0.7,"stddev":0.1,"count":3,"failed":0},"!setBreakpoints.errors":[],"setExceptionBreakpoints":{"operation":"setExceptionBreakpoints","totalTime":0.6,"max":0.6,"avg":0.6,"stddev":null,"count":1,"failed":0},"!setExceptionBreakpoints.errors":[],"configurationDone":{"operation":"configurationDone","totalTime":1,"max":1,"avg":1,"stddev":null,"count":1,"failed":0},"!configurationDone.errors":[]}}}},"level":0}
```

Thanks for you help!

**Comment by Gambero81** [comment]
same issue, on F5 the browser window remain blank with the spinning loading forever, the only way to go forward is to open the chrome web tools with F12, but in this way the debugger works only in chrome and no breakpoint are hit in visual studio.

@captainsafia this is a very priority issue, we need to wait until november release to be fixed?? as is the debug is impossible!

**Comment by ramious76** [comment]
Second on @Gambero81 .  I am using 16.7.3 and this is still an issue.  I have tried several of the fixes above and still no luck.

**Comment by axedaddy** [comment]
Same here.  Using 16.7.3 no luck on Edge(newest) and Chrome.  Can debug with docker but taking ~40s to start each time.

**Comment by captainsafia** [comment]
Hello folks -- thanks for your paitence here!

> I have solved this error when I am using Firefox.

We don't support debugging in Firefox. Our debugging experience relies on the Chrome Debugging Protocol, which is not supported by Firefox.

@spapaseit Thanks for sharing the logs. It looks like you are debugging a project that leverages auth. I'm investigating some issues related to debugging apps with an auth dependency. Stay tuned for updates on this front.

> same issue, on F5 the browser window remain blank with the spinning loading forever, the only way to go forward is to open the chrome web tools with F12, but in this way the debugger works only in chrome and no breakpoint are hit in visual studio.

This usually happens if the underlying DebugProxy isn't launched. We're working on improving the experience here so that the browser will close and alert the user instead of hanging.

**Comment by Gambero81** [comment]
> same issue, on F5 the browser window remain blank with the spinning loading forever
>
> This usually happens if the underlying DebugProxy isn't launched. We're working on improving the experience here so that the browser will close and alert the user instead of hanging.

@captainsafia  why the underlying DebugProxy isn't launched?
Have you found a workaround or there is a deep test i can do to help resolving the issue?

**Comment by captainsafia** [comment]
@Gambero81 If you are on 3.2, I recommend following [these steps](https://github.com/dotnet/aspnetcore/issues/22012#issuecomment-637117329) to get some helpful logs.

**Comment by markokamin** [comment]
i managed to get around the issue by killing the ".NET Host" process after each debugging session:
![image](https://user-images.githubusercontent.com/14233008/94110621-3691a500-fe43-11ea-94fe-6b8e8a7650d7.png)

Running VS 2019 Pro 16.8.0 Preview 3.0

**Comment by chassq** [comment]
This seems to still be an issue as of **Version 16.8.0 Preview 3.1 (Microsoft.AspNetCore Nuget Pkgs: 5.0.0-rc.1.20451.17)**. When using IISExpress launch settings profile the debugger does not seem to attach. However, when switching to the project profile (Kestrel) in launchsettings the debugger works as expected.

Tried removing the .vs file with no luck.

**Launch settings in Blazor wasm client and Blazor Server (Host).**
`{
    "iisSettings": {
      "windowsAuthentication": false,
      "anonymousAuthentication": true,
      "iisExpress": {
        "applicationUrl": "http://localhost:51544",
        "sslPort": 44305
      }
    },
    "profiles": {
      "IIS Express": {
        "commandName": "IISExpress",
        "launchBrowser": true,
        "launchUrl": "https://localhost:44305/",
        "environmentVariables": {
          "ASPNETCORE_ENVIRONMENT": "Development"
        }
      },
      "Ux.Client.Server": {
        "commandName": "Project",
        "launchBrowser": true,
        "inspectUri": "{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}",
        "applicationUrl": "https://localhost:5001;http://localhost:5000",
        "environmentVariables": {
          "ASPNETCORE_ENVIRONMENT": "Development"
        }
      }
    }
  }`

Microsoft Visual Studio Enterprise 2019 Preview
Version 16.8.0 Preview 3.1
VisualStudio.16.Preview/16.8.0-pre.3.1+30516.212
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Visual C++ 2019   00435-60000-00000-AA196
Microsoft Visual C++ 2019

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools 2019   16.8.442.31521
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2046102+2078c50b948ee8c2e2072fc69f5147b15e456b56
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.8.442.31521
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.8.442.31521
Azure App Service Tools v3.0.0

Azure Data Lake Node   1.0
This package contains the Data Lake integration nodes for Server Explorer.

Azure Data Lake Tools for Visual Studio   2.6.1000.1
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   16.8.442.31521
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.1000.1
Microsoft Azure Stream Analytics Tools for Visual Studio

Bundler & Minifier   2.8.396
Adds support for bundling and minifying JavaScript, CSS and HTML files in any project.

C# Tools   3.8.0-3.20458.6+b446afd34759150c2ea3b86ab794104e12e64db1
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

GhostDoc   20.2.20230.0
Generate XML Comments from your code, maintain clean and up-to-date documentation, produce help documentation in multiple formats, use intelligent source code Spell Checker in Visual Studio.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure HDInsight Azure Node   2.6.1000.1
HDInsight Node under Azure Node

Microsoft Azure Hive Query Language Service   2.6.1000.1
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   16.0
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.6.1000.1
Language service for Azure Stream Analytics

Microsoft Azure Stream Analytics Node   1.0
Azure Stream Analytics Node under Azure Node

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.30701.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.79+ge3567815aa.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   5.8.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62009.04150
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   16.0.20902.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.8.0-3.20458.6+b446afd34759150c2ea3b86ab794104e12e64db1
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.8.0-beta.20428.2+068ebd3c599bc5a47163a18b8b90d2fe5517186e
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions (Preview)   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

Visual Studio Tools for Kubernetes   1.0
Visual Studio Tools for Kubernetes

Web Compiler   1.12.394
Compiler for LESS, Sass and CoffeeScript files

**Comment by spapaseit** [comment]
I concur with @chassq, still an issue for me to on **Version 16.8.0 Preview 3.1 (Microsoft.AspNetCore Nuget Pkgs: 5.0.0-rc.1.20451.17)** as well

**Comment by captainsafia** [comment]
@spapaseit @chassq Can you share your log files. Should be under `%TEMPDIR%/visualstudio-chrome-debug.txt`after the failing debug session? Also, can you share the content of the `Output` tab in Visual Studio?

@markokamin This is a known issue and will be fixed in RC2.

**Comment by rasaconsulting-sandy** [comment]
I'm having this same issue - has anyone found a fix for it?  I can provide my logs too if it would help?

**Comment by captainsafia** [comment]
@rasaconsulting-sandy Can you open a new issue with the relevant details and logs?

**Comment by diogomota** [comment]
Hi - this may or may not be related but..

Debugging works flawlessly when the uris are set to **localhost:5000;localhost:5001**

However, I am hitting an Api with CORS enabled in the form of  *.domain.com    (I have no control over the cors policy of this api)

so my local uris need to be something like:   **localhost.domain.com:5000;localhost.domain.com:5001**  (Also have the correct hosts file change to make localhost.domain.com point to my machine (127.0.0.1))
This makes the API requests work fine but completely breaks the ability to hit breakpoints/debug.

(This is using the kestrel server)

Any idea what it might be?

**Comment by czf** [comment]
> A bit disappointing about resolution timeframe, here's what worked for me...
> I found this file:
> debuggerConfiguration-1859ae3b-3952-46c9-8065-7b07286175be.json
> under ....\MyApp.vs\MyApp\config
> it contains a single line
> {"chromiumDebuggerPort":60128}
> The file was a couple of weeks old, which struck me as odd, **anyhow i deleted the file.**


I had the same issue recently using VS 16.8.1 and saw @Will4000 steps.  After deleting that file, debugging start working again.

**Comment by tiagonmas** [comment]
Everything was working fine and suddenly I also started getting the error attaching the debuger to Edge. It works with Chrome or Firefox. 

Microsoft Visual Studio Enterprise 2019, Version 16.8.2
VisualStudio.16.Release/16.8.2+30717.126
Microsoft .NET Framework Version 4.8.04084
ASP.NET and Web Tools 2019   16.8.553.28003
ASP.NET Core Razor Language Services   16.1.0.2052803+84e121f1403378489b842e1797df2f3f5a49ac3c
ASP.NET Web Frameworks and Tools 2019   16.8.553.28003

**Comment by drewdelano** [comment]
I was having trouble with the "just hit F5 approach" but I'm able to attach to it from VS:
![image](https://user-images.githubusercontent.com/22156330/100533934-de6a7c80-31be-11eb-8bf1-2556a8df5ea4.png)

...after running the explicit command to enable remote debugging in Chrome 
![image](https://user-images.githubusercontent.com/22156330/100533980-53d64d00-31bf-11eb-8f01-ab94c6c24399.png)

**Comment by tolgaali** [comment]
The problem for me occurs if I select the Configure For HTTPS option when creating the project. As a workaround, for an HTTPS configured project changing isssettings->sslport to 0 and applicationUrl to only "http://localhost:nnnn" in launchsettings.json solved the issue. I think the debug adapter has an underlying security related problem.
